packages/odf/components/s3-iam/iam-user-details/UserOverviewPage.tsx

```diff
@@ -0,0 +1,151 @@
+import { CustomActionsToggle } from '../../s3-browser/objects-list';
+import { useFetchUserAndAccessKeys } from '../hooks/useFetchUserAndAccessKeys';
+import { NoobaaIAMContext } from '../iam-context';
+import { IAMAccessKeysDetails } from './iam-accesskeys-details/AccessKeysDetails';
+import { IAMTagsDetails } from './iam-tags-details/TagsDetails';
+
+const getUsersActionsItems = (
+  t: TFunction,
+  launcher: LaunchModal,
+  userName: string,
+  noobaaIAM: IAMCommands,
+  refreshTokens: () => void
+): IAction[] => [
+  {
+    title: t('Delete User'),
+    onClick: () =>
+      launcher(DeleteIAMUserModal as any, {
+        isOpen: true,
+        extraProps: {
+          userName,
+          noobaaIAM,
+          refreshTokens,
+        },
+      }),
+  },
+];
+
+const createUsersActions = (
+  t: TFunction,
+  fresh: boolean,
+  triggerRefresh: () => void,
+  launcher: ReturnType<typeof useModal>,
+  userName: string,
+  noobaaIAM: IAMCommands
+) => {
+  return (
+    <>
+      <Button
+        className="pf-v5-u-mr-md pf-v5-u-mb-xs"
+        variant={ButtonVariant.link}
+        icon={<BlueSyncIcon />}
+        onClick={triggerRefresh}
+        isDisabled={!fresh}
+        isInline
+      >
+        {t('Refresh')}
+      </Button>
+
+      <ActionsColumn
+        items={getUsersActionsItems(
+          t,
+          launcher,
+          userName,
+          noobaaIAM,
+          triggerRefresh
+        )}
+        actionsToggle={CustomActionsToggle}
+      />
+    </>
+  );
+};
+
+/**
+ * Fetches the userName from url params and renders the Tags and Accesskey Details Tab
+ */
+const UserOverview: React.FC<{}> = () => {
+  const { t } = useCustomTranslation();
+  const [fresh, triggerRefresh] = useRefresh();
+  const { resourceName: userName } = useParams();
+  const { noobaaIAM } = React.useContext(NoobaaIAMContext);
+
+  const launcher = useModal();
+
+  // Fetch user details and access keys
+  const { userDetails, iamAccessKeys, error, isLoading, refetch } =
+    useFetchUserAndAccessKeys(userName, noobaaIAM);
+  const navPages: TabPage[] = [
+    {
+      href: 'details',
+      name: t('Details'),
+      component: IAMTagsDetails,
+    },
+    {
+      href: 'accessKeys',
+      name: t('Access Keys'),
+      component: IAMAccessKeysDetails,
+    },
+  ];
+
+  const renderActions = () => () =>
+    createUsersActions(t, fresh, triggerRefresh, launcher, userName, noobaaIAM);
+
+  const breadcrumbs = [
+    {
+      name: t('Object Storage'),
+      path: '/odf/object-storage',
+    },
+    {
+      name: t('IAM'),
+      path: `/odf/object-storage/iam`,
+    },
+    {
+      name: t('User'),
+      path: '',
+    },
+  ];
+
+  const customData: IAMUserDetails = React.useMemo(
+    () => ({
+      fresh,
+      triggerRefresh,
+      userName,
+      noobaaIAM,
+      userDetails,
+      iamAccessKeys,
+      error,
+      isLoading,
+      refetch,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
```

Comment:
plz add deps to the list, I don't see any reason to skip all deps here...
especially `fresh` can change, in which case we need to update refs...
if `userDetails` and `iamAccessKeys` needs memoisation, we can do so in the respective hook...
```suggestion
    [<ADD_NEEDED_DEPS>]
```